library/pod-security-policy/seccomp/template.yaml

```diff
@@ -35,52 +35,220 @@ spec:
+        	file == allowed_files[_]
         }
 
-        input_wildcard_allowed(metadata) {
-            input.parameters.allowedProfiles[_] == "*"
+        # seccomp Localhost with wildcard
+        allowed_profile(profile, file, allowed) {
+        	profile == "Localhost"
+        	input_wildcard_allowed_files
+        	profile == allowed[_]
         }
 
-        input_container_allowed(metadata, container) {
-            not get_container_profile(metadata, container)
-            metadata.annotations["seccomp.security.alpha.kubernetes.io/pod"] == input.parameters.allowedProfiles[_]
+        # annotation localhost with wildcard
+        allowed_profile(profile, file, allowed) {
+        	"localhost/*" == allowed[_]
+        	startswith(profile, "localhost/")
         }
 
-        input_container_allowed(metadata, container) {
-            profile := get_container_profile(metadata, container)
-            profile == input.parameters.allowedProfiles[_]
+        # annotation localhost without wildcard
+        allowed_profile(profile, file, allowed) {
+        	startswith(profile, "localhost/")
+        	profile == allowed[_]
         }
 
-        get_container_profile(metadata, container) = profile {
-            value := metadata.annotations[key]
-            startswith(key, "container.seccomp.security.alpha.kubernetes.io/")
-            [prefix, name] := split(key, "/")
-            name == container.name
-            profile = value
+        # Localhost files from annotation scheme
+        get_annotation_localhost_files[file] {
+        	profile := input.parameters.allowedProfiles[_]
+        	startswith(profile, "localhost/")
+        	file := split(profile, "/")[1]
         }
 
-        input_containers[c] {
-            c := input.review.object.spec.containers[_]
+        # The profiles explicitly in the list
+        get_allowed_profiles[allowed] {
+        	allowed := input.parameters.allowedProfiles[_]
         }
-        input_containers[c] {
-            c := input.review.object.spec.initContainers[_]
+
+        # The simply translated profiles
+        get_allowed_profiles[allowed] {
+        	profile := input.parameters.allowedProfiles[_]
+        	not startswith(lower(profile), "localhost")
+        	allowed := naming_translation[profile][_]
+        }
+
+        # Seccomp Localhost to annotation translation
+        get_allowed_profiles[allowed] {
+        	profile := input.parameters.allowedProfiles[_]
+        	profile == "Localhost"
+        	file := object.get(input.parameters, "allowedLocalhostFiles", [])[_]
+        	allowed := sprintf("%v/%v", [naming_translation[profile][_], file])
+        }
+
+        # Annotation localhost to Seccomp translation
+        get_allowed_profiles[allowed] {
+        	profile := input.parameters.allowedProfiles[_]
+        	startswith(profile, "localhost")
+        	allowed := naming_translation.localhost[_]
```

Comment:
scratch: what happens to the file name?

Reply:
The file names are extracted in `get_annotation_localhost_files` when we verify the `localhostProfile` field.

Reply:
Sorry, I forgot to delete this comment, I was keeping it around so I wouldn't forget the thought.